src/object/srv_obj.c

```diff
@@ -1809,7 +1818,7 @@ obj_get_iods_offs_by_oid(daos_unit_oid_t uoid, struct obj_iod_array *iod_array,
 		}
 	}
 	if (oiod_nr > LOCAL_SKIP_BITS_NUM || *skips == NULL) {
-		D_ALLOC(*skips, roundup(oiod_nr / NBBY, 4));
```

Comment:
the original code seems no bug right? just it possibly will allocate 1 ~ 3 more bytes to be 4B aligned.

Reply:
"roundup(oiod_nr / NBBY, 4)" can be zero if `oiod_nr` is smaller than 8.

Reply:
Ah, right. could be bug when *skips == NULL

Reply:
Ah, right. could be bug when *skips == NULL